### PR TITLE
bufferlist: fix the bug of memory consolidation

### PR DESCRIPTION
**Pull request by YankunLi**
Signed-off-by: Yankun Li <lioveni99@gmail.com>

### CODE REVIEW

File: src/common/buffer.cc [hunk 1]
```diff
@@ -1826,7 +1826,7 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
     if (!_buffers.empty()) {
       ptr &l = _buffers.back();
       if (l.get_raw() == bp.get_raw() &&
-	  l.end() == bp.start() + off) {
+	  l.end() == off) {
```
Comment (tchaikov) [inline src/common/buffer.cc]: the `off` here is the offset inside of `bp` which start from `_off` in the backed raw data, while `l.end()` is the offset from the raw data. i don't think they are comparable.

Comment (YankunLi) [inline src/common/buffer.cc]: Thinks， but, I don't think so, the off here equal to append_buffer.end() - gap(the length of data that is append), so it is also the offset from the raw data.

Comment (YankunLi) [inline src/common/buffer.cc]: @tchaikov  If you are right, this call( #12247 ) must be error。

**Comment by tchaikov** [comment]
closing this PR, as #12247 is the right fix.

**Pull request closed without merging**